src/radical/pilot/agent/executing/popen.py

```diff
@@ -68,10 +65,59 @@ def initialize(self):
 
     # --------------------------------------------------------------------------
     #
-    def cancel_task(self, uid):
+    def get_task(self, tid):
+
+        return self._tasks.get(tid)
+
+
+    # --------------------------------------------------------------------------
+    #
+    def cancel_task(self, task):
```

Comment:
something wrong with states update - tmgr has the following states for cancelled tasks
```
state: "FAILED"
target_state: "CANCELED"
```

Reply:
```
$ cat agent_staging_output.0000.log | grep "m0.i0.s9.hello.2.0000"
1734567093.938 : agent_staging_output.0000 : 47102 : 140624563013184 : DEBUG    : task m0.i0.s9.hello.2.0000 skips staging: FAILED
1734567094.000 : agent_staging_output.0000 : 47102 : 140624563013184 : DEBUG    : task m0.i0.s9.hello.2.0000 skips staging: CANCELED
```
appears two times in staging output

Reply:
the following fix https://github.com/radical-cybertools/radical.pilot/commit/64493294bad4d6df0c0224389861809563abdf22 helped to overcome the issue (it is within the `project/impeccable` branch)